### ISSUE TITLE
Inlined asset doesn't compile correctly >=1.1.4

### ISSUE THREAD

**Issue by richhollis** [capistrano/rails#160]
Hi everyone

I have a Rails 3.2 app (3-2-stable) which has uses some inlined asset CSS in some layouts/views. e.g.

```
<style><%= Rails.application.assets['home.css'].to_s.html_safe %></style>
```

With 1.1.3 the assets:precompile step does take a good few seconds to do its thing but afterwards everything works as expected and has done for some time.

With 1.1.4 or 1.1.5 the inlined assets don't work as expected. For example, font-url entries just have the file name and don't have any compiled urls.

Rolling back to 1.1.3 fixes the problem.

Any idea what's causing the problem?

Many thanks
Rich

**Comment by mattbrictson** [comment]
Off the top of my head, 1.1.4 introduce a [bug that affected Rails 3 only](https://github.com/capistrano/rails/issues/142). But it should have been fixed in 1.1.5, so that must not be it.

Could you provide a bit more detail to help us troubleshoot? I can't guarantee I'll find the cause but maybe it will jump at out me.

* In the Capistrano logging output, is the `rake assets:compile` command, arguments, environment, etc. identical between 1.1.3 and 1.1.5? Pasting the output here would help.
* Did you upgrade and roll back any other gems at the same time as capistrano-rails? Sometimes issues like these can be triggered by an upgrade of sprockets or sprockets-rails.

**Comment by richhollis** [comment]
Hi Matt

Thanks for this. Yes - I did update all the gem dependencies so maybe there is something else creeping in. I'll roll back the gem updates and test the newer versions to see if the problem is still there. I'll also take a closer look at the logging output around the precompile and report back shortly.

Thanks
Rich

**Comment by mattbrictson** [comment]
I just ran into something similar myself, and confirmed it is due to the recently-released sprockets-rails 3.0.0. There is some more information here: https://github.com/rails/sprockets-rails/issues/294#issuecomment-166059398

I'm going to close this issue as I believe it is not capistrano-rails related. Please reopen if you find otherwise.

**Comment by richhollis** [comment]
Thanks @mattbrictson. 

My current sprockets version is 2.2.3 but I suspect something has changed around the use of Rails.application.assets - most likely returning nil in production which now might explain why the deployment went so fast because most likely it bailed during deploy/compilation.

I can see from the sprockets README the assets_manifest method was exposed in 3.0.0 which doesn't quite connect the dots, but having refactored my inline asset usage to use the actual compiled asset from the manifest.yml this problem no longer occurs. I can't access assets_manifest in rails-3-2-stable to try that out and see if it does work for inlined assets. 

Anyhow, not a problem with the capistrano-rails gem though. Thanks again for the quick response!
